install/scripts/install-spack-hipsycl.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+HIPSYCL_PKG_LLVM_VERSION_MAJOR=${HIPSYCL_PKG_LLVM_VERSION_MAJOR:-10}
+HIPSYCL_PKG_LLVM_VERSION_MINOR=${HIPSYCL_PKG_LLVM_VERSION_MINOR:-0}
+HIPSYCL_PKG_LLVM_VERSION_PATCH=${HIPSYCL_PKG_LLVM_VERSION_PATCH:-1}
+HIPSYCL_PKG_LLVM_REPO_BRANCH=${HIPSYCL_PKG_LLVM_REPO_BRANCH:-release/${HIPSYCL_PKG_LLVM_VERSION_MAJOR}.x}
+
+export HIPSYCL_INSTALL_PREFIX=${HIPSYCL_INSTALL_PREFIX:-/opt/hipSYCL/ }
+
+set -e
+HIPSYCL_BUILD_DIR=${HIPSYCL_BUILD_DIR:-/tmp/hipsycl-installer-sbalint}
+HIPSYCL_REPO_USER=${HIPSYCL_REPO_USER:-sbalint98}
+HIPSYCL_REPO_BRANCH=${HIPSYCL_REPO_BRANCH:-spack-packaging}
+HIPSYCL_WITH_CUDA=${HIPSYCL_WITH_CUDA:-OFF}
+HIPSYCL_WITH_ROCM=${HIPSYCL_WITH_ROCM:-ON}
+
+LLVM_INCLUDE_PATH=$HIPSYCL_INSTALL_PREFIX/llvm/llvm/lib/clang/${HIPSYCL_PKG_LLVM_VERSION_MAJOR}.\
+${HIPSYCL_PKG_LLVM_VERSION_MINOR}.\
+${HIPSYCL_PKG_LLVM_VERSION_PATCH}/include
+if [ -d "$HIPSYCL_BUILD_DIR" ]; then
+       read -p  "The build directory already exists, do you want to use $HIPSYCL_BUILD_DIR anyways?[y]" -n 1 -r
+       echo 
+       if [[ ! $REPLY =~ ^[Yy]$ ]]; then
+              echo "Please specify a different directory, exiting"
+              [[ "$0" = "$BASH_SOURCE" ]] && exit 1 || return 1
+       else
+              echo "Using the exisiting directory"
+       fi
+else
+echo "Cloning hipSYCL"
+git clone --recurse-submodules -b $HIPSYCL_REPO_BRANCH https://github.com/$HIPSYCL_REPO_USER/hipSYCL $HIPSYCL_BUILD_DIR
+
+fi
+
+mkdir -p $HIPSYCL_BUILD_DIR/build
+cd $HIPSYCL_BUILD_DIR/build
+
+# We need the llvm module to be loaded in order to be able to find the openmp rt
+export SPACK_ROOT=/root/spack
```

Comment:
Is it possible to formulate it such that it works by default also for non-privileged users who wish to install on bare metal (i.e. not building singularity image)?

Reply:
I don't see right now how this could work well. I think these spack-install scripts are not intended to be used by a regular user. We have to do some hacking (changing where spack installs the packages) to have the different dependencies in separate folders so that they can be packaged. A user would probably be happy if all the dependencies are installed in the same directory. However, I think that a separate script would be handier for that purpose. Maybe these scripts should be moved to `devops/repo`. 

Ultimately I think all of this should be part of the CI workflows, but for that, we probably need to update jarvis, since the singularity docs recommend 3.18 kernel version for --fake-root. (jarvis has 3.10)